substrate/state-machine/src/lib.rs

```diff
@@ -113,10 +112,23 @@ impl OverlayedChanges {
 		}
 	}
 
-	/// Drain prospective changes to an iterator.
-	pub fn drain(&mut self) -> Drain<Vec<u8>, Option<Vec<u8>>> {
+	/// Drain committed changes to an iterator.
+	///
+	/// Panics:
+	/// Will panic if there are any uncommitted prospective changes.
```

Comment:
Is "prospective" sort of like "pending"?

Reply:
Yes, those are changes that can still be easily discarded. You can see example of usage inside block builder:
1. We run a transaction
2. It produces a set of prospective changes
3. If we detect that it's somehow invalid we discard the prospective changes
4. If we accept the transaction we commit prospective changes.